### DISCUSSION TITLE
Hit graphQL endpoint

### DISCUSSION THREAD

**Discussion by jerem-eng** [pimcore/pimcore#9428] [General]
Hello there, 

I'm new in pimcore, and i have some trouble with de API endpoint in graphQL.

I'm tying to get some product with the datahub. so i configure the endpoint in graphQL, i get my product in the graphQL view in pimcore. 

but when i try to hit the endpoint with insomnia, i have a response 200 with a blank page. there is one button "enter token", when i click i have a field but i cant validate anything.

i try to pass token whith bearer token, no succes, i try to pass token by the url like that www.myurl.fr/products?apikey=myApiKey but no more success. 

i didn't find anything in pimcore senior courses about datahub

thank you for you help :-)

**Comment by Jimerson21** [comment]
Hello jerem-eng,

the calls to the Api are made in this way:

http://mydomain/pimcore-graphql-webservices/MyEndpoint?apikey=XXXXXX

https://pimcore.com/docs/data-hub/current/GraphQL/Configuration/Customize_Endpoint_URL.html

**Reply by jerem-eng** [reply]
Hello Jimerson21, 

thanks for your reply, yes i call my api like this 
<img width="594" alt="Capture d’écran 2021-06-17 à 10 06 40" src="https://user-images.githubusercontent.com/85865177/122357211-d2c08680-cf53-11eb-96a6-1f0bcb4ce98c.png">

when i hit the endpoint with insomia for exemple i get this page 
<img width="622" alt="Capture d’écran 2021-06-17 à 10 07 52" src="https://user-images.githubusercontent.com/85865177/122357409-04d1e880-cf54-11eb-8787-71899d6f5d79.png">
<img width="624" alt="Capture d’écran 2021-06-17 à 10 08 07" src="https://user-images.githubusercontent.com/85865177/122357414-069bac00-cf54-11eb-926d-4d8e408eecab.png">

i tried with the pimcore demo, but i have the same result.
I think i miss something with the token but i don't know what

**Reply by Jimerson21** [reply]
I'm afraid you can't query the demo Api, without the token.

But if you create your own endpoint, you don't need this token, the api key is enough.

![image](https://user-images.githubusercontent.com/58362749/122399200-9176a980-cf48-11eb-9850-5abe79066a1a.png)
